plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -752,11 +753,12 @@ private ConnectorTableMetadata doGetTableMetadata(ConnectorSession session, Sche
                 return Stream.empty();
             }
             catch (TrinoException e) {
+                ErrorCode errorCode = e.getErrorCode();
                 // Skip this table if there's a failure due to Hive, a bad Serde, or bad metadata
-                if (!e.getErrorCode().getType().equals(ErrorType.EXTERNAL)) {
-                    throw e;
+                if (errorCode.getType().equals(ErrorType.EXTERNAL) || errorCode.equals(UNSUPPORTED_TABLE_TYPE.toErrorCode())) {
```

Comment:
@findinpath explained me what the problem is, related to https://github.com/trinodb/trino/pull/11539
and i decided what the fix should be. @findinpath will raise a PR shortly